components/org.wso2.micro.integrator.ndatasource.core/src/main/java/org/wso2/micro/integrator/ndatasource/core/utils/DataSourceUtils.java

```diff
@@ -141,26 +147,53 @@ public static boolean nullAllowEquals(Object lhs, Object rhs) {
 		}
 		return lhs.equals(rhs);
 	}
-	
-	public static String elementToString(Element element) {
+
+	/**
+	 * This method will return the serialized Data source.
+	 *
+	 * @param element           Data Source Configuration
+	 * @param isUiSerialization Sensitive information will be masked if isUiSerialization is true
+	 * @return String Serialized Data source
+	 */
+	public static String elementToString(Element element, boolean isUiSerialization) {
+
 		try {
 			if (element == null) {
                                 /* return an empty string because, the other way around works the same,
                                 where if we give a empty string as the XML, we get a null element
                                 from "stringToElement" */
 				return "";
 			}
-            Document document = element.getOwnerDocument();
-            DOMImplementationLS domImplLS = (DOMImplementationLS) document.getImplementation();
-            LSSerializer serializer = domImplLS.createLSSerializer();
-            //by default its true, so set it to false to get String without xml-declaration
-            serializer.getDomConfig().setParameter(XML_DECLARATION, false);
-            return serializer.writeToString(element);
+			Element updatedElement;
+			if (isUiSerialization) {
+				updatedElement = (Element) element.cloneNode(true);
+				for (Node child = updatedElement.getFirstChild(); child != null; child = child.getNextSibling()) {
+					if (child instanceof Element && PWD_ELEMENT.equals(child.getNodeName())) {
+						child.getFirstChild().setNodeValue(PWD_MASKED_VALUE);
+					}
+					if (child instanceof Element && URL_ELEMENT.equals(child.getNodeName())) {
+						child.getFirstChild().setNodeValue(maskURLPassword(child.getFirstChild().getNodeValue()));
+					}
+				}
+			} else {
+				updatedElement = element;
+			}
```

Comment:
shall we refactor this to avoid if and else

Reply:
Fixed with 36c2b7f